common/src/main/java/com/wynntils/models/character/RankModel.java

```diff
@@ -0,0 +1,146 @@
+package com.wynntils.models.character;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Model;
+import com.wynntils.core.persisted.Persisted;
+import com.wynntils.core.persisted.storage.Storage;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.handlers.chat.event.ChatMessageReceivedEvent;
+import com.wynntils.handlers.container.scriptedquery.QueryBuilder;
+import com.wynntils.handlers.container.scriptedquery.QueryStep;
+import com.wynntils.handlers.container.scriptedquery.ScriptedContainerQuery;
+import com.wynntils.handlers.container.type.ContainerContent;
+import com.wynntils.models.containers.ContainerModel;
+import com.wynntils.models.worlds.event.WorldStateEvent;
+import com.wynntils.models.worlds.type.WorldState;
+import com.wynntils.utils.mc.LoreUtils;
+import com.wynntils.utils.type.ConfirmedBoolean;
+import com.wynntils.utils.wynn.InventoryUtils;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.item.ItemStack;
+import net.neoforged.bus.api.EventPriority;
+import net.neoforged.bus.api.SubscribeEvent;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class RankModel extends Model {
+    // Test in CharacterModel_VETERAN_PATTERN
```

Comment:
What if we shoot higher and make this `AccountModel`? That way, we have a `CharacterModel` for the current class, and an `AccountModel` for global account knowledge. I think this is going to be useful soon enough.

Reply:
And then, might as well give the `AccountModel` a new package.